### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by willoliveira**
### Feature request summary

This feature includes CircleCI to build and deliver design token documentation

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const gulp = require('gulp');
```
Comment (eltongarbin) [inline gulpfile.js:1]: For organization purposes I think it should better if this file and others about build process (android folder / ios folder) must be in the scripts folder.

See examples about this pattern:
https://github.com/facebook/react/tree/master/scripts
https://github.com/facebook/jest/tree/master/scripts

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      - build-web:
+          requires:
+            - build-setup
+      - build-docs:
+          requires:
+            - build-setup
+      - deploy-docs:
+          requires:
+            - build-setup
+            - build-docs
```
Comment (eltongarbin) [inline .circleci/config.yml:81]: I think we don't need to trigger a workflow on a schedule like that.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
```
Comment (eltongarbin) [inline .circleci/config.yml:94]: Change to trigger deploy-docs job only in master branch

```suggestion
  build:
    jobs:
      - build-setup
      - build-web:
          requires:
            - build-setup
      - build-docs:
          requires:
            - build-setup
      - deploy-docs:
          requires:
            - build-docs
          filters:
            branches:
              only: master
```

Comment (eltongarbin) [inline .circleci/config.yml:94]: Change to trigger deploy-docs only in master branch

```suggestion
      - deploy-docs:
          requires:
            - build-docs
          filters:
            branches:
              only: master            
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+version: 2.1
+orbs:
+  node: circleci/node@1.1.6
```
Comment (eltongarbin) [inline .circleci/config.yml:3]:
```suggestion
  node: circleci/node@3.0.0
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+jobs:
+  build-setup:
+    executor:
+      name: node/default
```
Comment (eltongarbin) [inline .circleci/config.yml:7]:
```suggestion
    executor: node/default
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+          steps:
+            - run:
+                name: Install Yarn package manager
+                command: curl -o- -L https://yarnpkg.com/install.sh | bash
+            - run:
+                name: Install javascript dependencies (using Yarn)
+                command: yarn install --ignore-engines
+  build-web:
+    executor:
+      name: node/default
```
Comment (eltongarbin) [inline .circleci/config.yml:20]:
```suggestion
    executor: node/default
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    steps:
+      - checkout
+      - node/with-cache:
+          steps:
+            - run:
+                name: Build design tokens for web
+                command: yarn web
+  build-docs:
+    executor:
+      name: node/default
```
Comment (eltongarbin) [inline .circleci/config.yml:30]:
```suggestion
    executor: node/default
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    steps:
+      - checkout
+      - node/with-cache:
+          steps:
+            - run:
+                name: Build design tokens for web
```
Comment (eltongarbin) [inline .circleci/config.yml:36]:
```suggestion
                name: Build design-tokens docs
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+                command: yarn docs
+            - persist_to_workspace:
+                root: dist
+                paths:
+                  - docs
+  deploy-docs:
+    docker:
+      - image: node:8.10.0
```
[the remark this answers is not in the document]
Comment (willoliveira) [reply .circleci/config.yml:44]: Gh pages requires this Node version

**Pull request merged**